common/streams/src/stream_limit.rs

```diff
@@ -14,38 +14,64 @@ use crate::SendableDataBlockStream;
 
 pub struct LimitStream {
     input: SendableDataBlockStream,
-    limit: usize,
+    limit: Option<usize>,
```

Comment:
Maybe it should be `offset: Option<usize>`?

SQL LIMIT syntax:
SELECT * FROM `table` [LIMIT [offset_index, ] limit_size]

Reply:
It seems that there is no any differences between offset 0 and no offset clause, so I omit it.

Reply:
the offset and limit is just like Stream::size_hint(), which use usize representing lower bound and Option<usize> representing upper bound.